src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -459,6 +460,14 @@ def create_user(self, username, password, name, email, enabled=True,
                         pwd_update_required=pwd_update_required)
             self.users[username] = user
             return user
+    
+    def save_attempt(self, username, attempt):
+        self.attempt[username] = attempt
```

Comment:
I think the assignment operation should be performed under threading lock

```suggestion
        with self.lock:
             self.attempt[username] = attempt
```

Reply:
Yes. I've made the changes locally. Also changed the logic a bit. I'll push it in soon.